### ISSUE TITLE
ansible-doc doesn't support ANSIBLE_METADATA in YAML 

### ISSUE THREAD

**Issue by jtyr** [ansible/ansible#19867]
##### ISSUE TYPE

Bug Report

##### COMPONENT NAME

`ansible-doc`

##### ANSIBLE VERSION
```
ansible 2.2.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

##### CONFIGURATION

Tested with `devel` branch.

##### OS / ENVIRONMENT

Running on Linux.

##### SUMMARY

When defining `ANSIBLE_METADATA` in YAML format, the testing fails on `ansible-doc` which is not able to read `dict` from the metadata.

##### STEPS TO REPRODUCE

Insert the following metadata into a module and run the test:

```
ANSIBLE_METADATA = """
status:
  - preview
supported_by: community
version: '1.0'
"""
```

##### EXPECTED RESULTS

Test passes.

##### ACTUAL RESULTS

Test failing:

```
$ tox -c test/runner/tox.ini -e py27 -- /tmp/ansible/test/runner/test.py sanity -v --test ansible-doc --coverage --require lib/ansible/modules/network/ldap_attr.py --color yes --requirements --python 2.7
py27 installed: ansible==2.3.0,astroid==1.4.9,backports.functools-lru-cache==1.3,configparser==3.5.0,coverage==4.3.1,funcsigs==1.0.2,isort==4.2.5,Jinja2==2.8.1,lazy-object-proxy==1.2.2,MarkupSafe==0.23,mccabe==0.5.3,mock==2.0.0,pbr==1.10.0,pylint==1.6.4,PyYAML==3.12,six==1.10.0,voluptuous==0.9.3,wrapt==1.10.8,yamllint==1.6.0
py27 runtests: PYTHONHASHSEED='1455696754'
py27 runtests: commands[0] | /tmp/ansible/test/runner/test.py sanity -v --test ansible-doc --coverage --require lib/ansible/modules/network/ldap_attr.py --color yes --requirements --python 2.7
Run command: pip install --disable-pip-version-check -r test/runner/requirements/sanity.txt -c test/runner/requirements/constraints.txt coverage
Requirement already satisfied: coverage>=4.2 in ./test/runner/.tox/py27/lib/python2.7/site-packages (from -c test/runner/requirements/constraints.txt (line 1))
Requirement already satisfied: jinja2 in ./test/runner/.tox/py27/lib/python2.7/site-packages (from -r test/runner/requirements/sanity.txt (line 1))
Requirement already satisfied: mock in ./test/runner/.tox/py27/lib/python2.7/site-packages (from -r test/runner/requirements/sanity.txt (line 2))
Requirement already satisfied: pylint in ./test/runner/.tox/py27/lib/python2.7/site-packages (from -r test/runner/requirements/sanity.txt (line 3))
Requirement already satisfied: voluptuous in ./test/runner/.tox/py27/lib/python2.7/site-packages (from -r test/runner/requirements/sanity.txt (line 4))
Requirement already satisfied: yamllint in ./test/runner/.tox/py27/lib/python2.7/site-packages (from -r test/runner/requirements/sanity.txt (line 5))
Requirement already satisfied: MarkupSafe in ./test/runner/.tox/py27/lib/python2.7/site-packages (from jinja2->-r test/runner/requirements/sanity.txt (line 1))
Requirement already satisfied: funcsigs>=1; python_version < "3.3" in ./test/runner/.tox/py27/lib/python2.7/site-packages (from mock->-r test/runner/requirements/sanity.txt (line 2))
Requirement already satisfied: six>=1.9 in ./test/runner/.tox/py27/lib/python2.7/site-packages (from mock->-r test/runner/requirements/sanity.txt (line 2))
Requirement already satisfied: pbr>=0.11 in ./test/runner/.tox/py27/lib/python2.7/site-packages (from mock->-r test/runner/requirements/sanity.txt (line 2))
Requirement already satisfied: isort>=4.2.5 in ./test/runner/.tox/py27/lib/python2.7/site-packages (from pylint->-r test/runner/requirements/sanity.txt (line 3))
Requirement already satisfied: mccabe in ./test/runner/.tox/py27/lib/python2.7/site-packages (from pylint->-r test/runner/requirements/sanity.txt (line 3))
Requirement already satisfied: configparser; python_version == "2.7" in ./test/runner/.tox/py27/lib/python2.7/site-packages (from pylint->-r test/runner/requirements/sanity.txt (line 3))
Requirement already satisfied: backports.functools-lru-cache; python_version == "2.7" in ./test/runner/.tox/py27/lib/python2.7/site-packages (from pylint->-r test/runner/requirements/sanity.txt (line 3))
Requirement already satisfied: astroid<1.5.0,>=1.4.5 in ./test/runner/.tox/py27/lib/python2.7/site-packages (from pylint->-r test/runner/requirements/sanity.txt (line 3))
Requirement already satisfied: setuptools>=0.6b1 in ./test/runner/.tox/py27/lib/python2.7/site-packages (from voluptuous->-r test/runner/requirements/sanity.txt (line 4))
Requirement already satisfied: pyyaml in ./test/runner/.tox/py27/lib/python2.7/site-packages (from yamllint->-r test/runner/requirements/sanity.txt (line 5))
Requirement already satisfied: wrapt in ./test/runner/.tox/py27/lib/python2.7/site-packages (from astroid<1.5.0,>=1.4.5->pylint->-r test/runner/requirements/sanity.txt (line 3))
Requirement already satisfied: lazy-object-proxy in ./test/runner/.tox/py27/lib/python2.7/site-packages (from astroid<1.5.0,>=1.4.5->pylint->-r test/runner/requirements/sanity.txt (line 3))
Sanity check using ansible-doc with Python 2.7
Run command: ansible-doc ldap_attr
ERROR: Output on stderr from ansible-doc is considered an error.
ERROR: Command "ansible-doc ldap_attr" returned exit status 0.
>>> Standard Error
 [WARNING]: Non-dict metadata detected in /tmp/ansible/lib/ansible/modules/network/ldap_attr.py, skipping

ERROR: InvocationError: '/tmp/ansible/test/runner/test.py sanity -v --test ansible-doc --coverage --require lib/ansible/modules/network/ldap_attr.py --color yes --requirements --python 2.7'
__________________________________________________________________________________________________________________ summary __________________________________________________________________________________________________________________
ERROR:   py27: commands failed
```

**Comment by sivel** [comment]
I do believe this is expected.  `ANSIBLE_METADATA` must be supplied as a dictionary, and not in a YAML string format like the other pieces of documentation.

The warning indicates this as well.

**Comment by sivel** [comment]
I will note that this was a conscious decision, to require the data to be a dict, and discussed during the module repo merge meeting.

**Comment by jtyr** [comment]
@sivel but as per the [proposal](https://github.com/ansible/proposals/issues/30), the YAML format should be supported. @bcoca acknowledged the bug in the IRC channel.

**Comment by sivel** [comment]
The proposal did give a YAML example.  However in the meeting where we made the final decision, we decided to not support YAML.

My feeling is that, although this diverges from the other places where we use YAML strings, if we add support for YAML strings, we should remove support for it being a dict. We should only support 1 ideally for sake of standardization.

I understand it is relatively trivial to add, but I do not count this as a bug.

**Comment by nitzmahone** [comment]
Yeah, the initial impl only supports dicts, since that was the minimum we needed to translate/augment the initial metadata (supported_by, etc). Since docs are currently only Python ATM, we've deferred YAML support indefinitely, until there's a good reason for it. IIRC, we went with the dict because we've already gone to the trouble to parse/compile the Python, so it's already done vs needing to then read/parse YAML inside that (thus slowing down ansible-doc and anything else that needs the metadata).
